db/sqldb/pacta_version.go

```diff
@@ -0,0 +1,193 @@
+	rows, err := d.query(tx, `
+		SELECT `+pactaVersionSelectColumns+`
+		FROM pacta_version 
+		WHERE is_default;`)
+	if err != nil {
+		return nil, fmt.Errorf("querying primary pacta_version: %w", err)
+	}
+	pvs, err := rowsToPACTAVersions(rows)
+	if err != nil {
+		return nil, fmt.Errorf("translating rows to pacta versions: %w", err)
+	}
+	return exactlyOne("pacta_version", "is_default", pvs)
+}
+
+func (d *DB) PACTAVersions(tx db.Tx) ([]*pacta.PACTAVersion, error) {
+	rows, err := d.query(tx, `
+		SELECT `+pactaVersionSelectColumns+`
+		FROM pacta_version;`)
+	if err != nil {
+		return nil, fmt.Errorf("querying pacta_version: %w", err)
+	}
+	pvs, err := allRows("pacta_version", rows, rowToPACTAVersion)
+	if err != nil {
+		return nil, fmt.Errorf("translating rows to pacta versions: %w", err)
+	}
+	return pvs, nil
+}
+
+func (d *DB) CreatePACTAVersion(tx db.Tx, pv *pacta.PACTAVersion) (pacta.PACTAVersionID, error) {
+	if err := validatePACTAVersionForCreation(pv); err != nil {
+		return "", fmt.Errorf("failed pacta_version validation: %w", err)
+	}
+	id := pacta.PACTAVersionID(d.randomID(pactaVersionIDNamespace))
+	createdAt := time.Now()
+	err := d.exec(tx, `
+		INSERT INTO pacta_version 
+			(id, name, description, digest, created_at, is_default)
+			VALUES
+			($1, $2, $3, $4, $5, $6);
+	`, id, pv.Name, pv.Description, pv.Digest, createdAt, nil)
+	if err != nil {
+		return "", fmt.Errorf("creating pacta_version: %w", err)
+	}
+	return id, nil
+}
+
+func (d *DB) SetDefaultPACTAVersion(tx db.Tx, id pacta.PACTAVersionID) error {
+	err := d.RunOrContinueTransaction(tx, func(db.Tx) error {
+		err := d.exec(tx, `UPDATE pacta_version SET is_default = NULL;`)
+		if err != nil {
+			return fmt.Errorf("setting all pacta_versions is_default to null: %w", err)
+		}
+		err = d.exec(tx, `UPDATE pacta_version SET is_default = TRUE WHERE id = $1;`, id)
+		if err != nil {
+			return fmt.Errorf("setting pacta version with id %q is_default: %w", id, err)
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("updating primary pacta_version: %w", err)
+	}
+	return nil
+}
+
+func (d *DB) UpdatePACTAVersion(tx db.Tx, id pacta.PACTAVersionID, mutations ...db.UpdatePACTAVersionFn) error {
+	err := d.RunOrContinueTransaction(tx, func(db.Tx) error {
+		pv, err := d.PACTAVersion(tx, id)
+		if err != nil {
+			return fmt.Errorf("reading pacta_version: %w", err)
+		}
+		for i, m := range mutations {
+			err := m(pv)
+			if err != nil {
+				return fmt.Errorf("running %d-th mutation: %w", i, err)
+			}
+		}
+		err = d.putPACTAVersion(tx, pv)
+		if err != nil {
+			return fmt.Errorf("putting pacta_version: %w", err)
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("updating pacta_version: %w", err)
+	}
+	return nil
+}
+
+func (d *DB) DeletePACTAVersion(tx db.Tx, id pacta.PACTAVersionID) error {
+	err := d.RunOrContinueTransaction(tx, func(db.Tx) error {
+		// TODO - include appropriate handling for deleting a pacta_version that is used.
```

Comment:
Or disallow it (i.e. current behavior)? Not sure what the implications are in either case

Reply:
Yeah I have to talk with Hodie + Alex about it. I'll formalize as a TODO.